File: Creational/Builder/Java/ConcreteBuilder2.java

public class ConcreteBuilder2 extends AbstractBuilder{

	@Override
	public void reset() {
		this.abstractProduct = new ConcreteProduct2();
		
	}

	@Override
	public void buildStepA() {
		this.abstractProduct.stepA2();
	}

	@Override
	public void buildStepB() {
		this.abstractProduct.stepB2();
		
	}

	@Override
	public void buildStepC() {
		this.abstractProduct.stepC2();
	}

}

File: Creational/Builder/Kotlin/ConcreteBuilder2.kt
class ConcreteBuilder2: AbstractBuilder() {

    override fun reset() {
        this.abstractProduct = ConcreteProduct2()
    }

    override fun buildStepA() {
        this.abstractProduct.stepA2()
    }

    override fun buildStepB() {
        this.abstractProduct.stepB2()
    }

    override fun buildStepC() {
        this.abstractProduct.stepC2()
    }

}